### PR TITLE
Mods batch#2

### LINKED ISSUE #47: As a <user> I want <a larger tag input window> so that <I can see the whole tag when entering it>.
@JKHeadley commented on [Wed Aug 12 2020](https://github.com/cusac/mbt.guide-backend/issues/26)

From Helene:

>Also, by the way, the window to write the tags is very small, too small, meaning if the tag contains more than one word or the word is long, you can't see what you have written. It would be great to have that window at least one finger long such that, at least, you can write "Double slit experiment" for example and see all the words.

### PR DESCRIPTION
**Pull request by oormicreations**
Branch : mods-batch#2
20-Jan-2021

On add segment, start time is set to end time of last segment
File: src/routes/Edit/VideoSplitter.tsx Line:128-134

Tag input is visible and centered with a bigger font, borders etc
File : src/App.css Line:79

Removed extra "searching" from the message. Added a tip.
File : src/routes/Home/index.tsx Line:497

Video description is collapsible and looks pretty with line breaks etc.
File : src/routes/Home/index.tsx Line:292

Header shows a welcome message and user name
File : src/components/AppHeader.tsx Line:84

20-Jan-2021

img changed to Image with link as onclick was not working
File : src/components/AppHeader.tsx Line:62

images folder - to avoid many copies of images
File : src/images

New component - landing page
File : src/components/LandingPage.tsx 

Feature - Landing Page (ya it is hacky based on the default video id)
File : src/routes/Home/index.tsx Line:319 (and many)

Help tips
File : src/routes/Edit/VideoSplitter.tsx Line:532, 590

### CODE REVIEW

**Comment by JKHeadley** [comment]
@oormicreations Just a general note, it looks like many of these changes are the same as your previous PR. Generally in my experience it is a cleaner approach to continue committing to a single branch until it is either merged or specifically put into the "icebox" for later attention.

So the flow could be (approximately):
- PR is submitted
- PR is reviewed and changes are requested                <-- this is where we left off on the last PR
- More edits are made to the same branch
- The processed continues until the changes are accepted
- The branch is merged
- A new branch is created for the next set of changes.

This way there is less redundancy between branches which makes them easier to merge and less branches in general.

Maybe we can hop on a zoom chat soon to talk through more details on workflows. As always, thanks for all the work!

**Comment by oormicreations** [comment]
@JKHeadley Sure, from now on I will update this branch, till it is merged.
Probably I misunderstood the instruction "send your changes in a new branch". :smile: 
Thank you.
Hopefully it will not show the old changes once the first branch is merged.

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,4 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+changelog-sanjeev.txt
```
Comment (JKHeadley) [inline .gitignore]: Adding ignores for personal files can clutter up the .gitignore file. You might be able to instead add them to your global .gitignore (not the project file). Or store them in a directory outside of the project. If you don't mind please revert this change.

File: changelog-sanjeev.txt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+File : src/images
+
+New component - landing page
+File : src/components/LandingPage.tsx 
+
+Feature - Landing Page (ya it is hacky based on the default video id)
+File : src/routes/Home/index.tsx Line:319 (and many)
+
+Help tips
+File : src/routes/Edit/VideoSplitter.tsx Line:532, 590
```
Comment (JKHeadley) [inline changelog-sanjeev.txt]: Since this info is in the PR description, lets not commit this file to the repo to prevent clutter.

File: src/App.css [hunk 3]
```diff
@@ -75,3 +76,50 @@
...
+  margin-bottom: 6px;
+  margin-top: 1px;
+  padding: 3px;
+  width: 300px;
+  text-align: center;
+  border-width: 1px;
+  border-color: grey;
+  border-style: dashed;
+  border-radius: 3px;
+}
```
Comment (JKHeadley) [inline src/App.css:104]: This looks much better, thank you!

File: src/routes/Edit/VideoSegmentItem.tsx [hunk 4]
```diff
@@ -33,6 +33,7 @@ const VideoSegmentItem = ({
               style={{ margin: 0 }}
               circular
               icon={canEdit ? 'edit' : 'lock'}
+              color={active ? 'green' : 'grey'}
```
Comment (JKHeadley) [inline src/routes/Edit/VideoSegmentItem.tsx:36]: Porting over this comment:

https://github.com/cusac/mbt.guide/pull/74#discussion_r559752842

File: src/routes/Edit/VideoSplitter.tsx [hunk 5]
```diff
@@ -125,11 +125,14 @@ const VideoSplitter = ({
...
     const newId = uuid();
+    const hasNoCurSeg = currentSegment === undefined;
+    const lastSegEnd = hasNoCurSeg ? '0' : (currentSegment as any).end;
+
     newSegments.push({
       segmentId: newId,
       video: (video as any)._id,
-      start: duration * 0.25,
-      end: duration * 0.75,
+      start: Number(lastSegEnd), //duration * 0.25,
```
Comment (JKHeadley) [inline src/routes/Edit/VideoSplitter.tsx:138]: I definitely understand the reasoning behind this logic. I think it would be goo to discuss some context that will affect this.

We can discuss in more detail later, but in brief the context to keep in mind is that once this site is open to the public, any individual will be welcome to create a segment. These individuals will not necessarily be "parsing a video" in a linear fashion as the volunteers have been doing, but rather adding a segment at some arbitrary position in the video.  In this context it is more consistent to set the start and end at the same point for each new segment.

I think your change is good for now since the site is not public and this change will ease the work of the volunteers. However moving forward I think this should be moved over to the user's profile settings (to be created) so that users can customize this setting.

File: src/routes/Edit/VideoSplitter.tsx [hunk 6]
```diff
@@ -527,6 +529,14 @@ const VideoSplitter = ({
                     onChange={(event: any) => updateEnd(index, event.target.value)}
                   />
                 </Grid.Column>
+                <Grid.Column
+                  verticalAlign="middle"
+                  style={{ textAlign: 'right', color: 'grey' }}
+                  width={6}
+                >
+                  <Icon name="help circle" />
+                  Tip : Type e.g. 012030 without : or spaces to enter 01:20:30.
```
Comment (JKHeadley) [inline src/routes/Edit/VideoSplitter.tsx:536]: These tips are great. Could we make sure both tips are the same color? Currently the top is grey and the bottom is black.

File: src/routes/Home/index.tsx [hunk 7]
```diff
@@ -287,6 +290,17 @@ const Home = ({ videoId }: { videoId: string }) => {
     utils.history.push(`/edit/${videoId}`);
   };
 
+  const descText = selectedVideo ? (selectedVideo as any).snippet.description : 'None';
```
Comment (JKHeadley) [inline src/routes/Home/index.tsx]: The `(selectedVideo as any)` is not necessary here. Instead lets use:

`const descText = selectedVideo ? selectedVideo.snippet.description : 'None';`

In general we want to avoid using the `any` type as much as possible as it weakens the type safety of the code.

Comment (JKHeadley) [reply src/routes/Home/index.tsx]: Note: Just to clarify, most of the `(<variable> as any)` code segments are there as a temporary fix from a global conversion from javascript to typescript in an earlier effort. My idea was to slowly convert to more strongly typed statements as development continued, or to make a single large effort to convert them all at once.

File: src/routes/Home/index.tsx [hunk 8]
```diff
@@ -300,30 +314,52 @@ const Home = ({ videoId }: { videoId: string }) => {
...
+                    <LandingPage user={currentUser ? (currentUser as any).firstName : 'Guest'} />
+                  ) : (
+                    <div>
+                      <div className="ui embed">
+                        <iframe src={videoSrc} allowFullScreen title="Video player" />
+                      </div>
+                      <div className="videodesc">
+                        <h3>{(selectedVideo as any).snippet.title}</h3>
+
+                        <p>{'Date Published : ' + (selectedVideo as any).snippet.publishedAt}</p>
```
Comment (JKHeadley) [inline src/routes/Home/index.tsx]: We will need to use a date formatting tool to correctly display this, as currently it shows, Ex:

`Date Published : 2021-01-19T17:26:26Z`

File: src/routes/Home/index.tsx [hunk 9]
```diff
@@ -300,30 +314,52 @@ const Home = ({ videoId }: { videoId: string }) => {
...
+                            color="teal"
+                            floated="right"
+                            size="mini"
+                            onClick={() => {
+                              setReadMore(!readMore);
+                            }}
+                          >
+                            {moreButtonName}
+                            <Icon className={readMore ? 'left arrow' : 'right arrow'} />
+                          </Button>
```
Comment (JKHeadley) [inline src/routes/Home/index.tsx:340]: This is a very nice improvement.

The functionality is very nice. I'm wondering if there might be a simpler way to represent the buttons. Something like how youtube displays them would be nice. Ex:

Screen Shot 2021-01-23 at 4 58 53 PM

Screen Shot 2021-01-23 at 4 59 04 PM

I like these because they aren't very "button like" so they are more subtle, while still being very clear. They also fit nicely right below the text. What do you think?

File: src/routes/Home/index.tsx [hunk 10]
```diff
@@ -287,6 +290,17 @@ const Home = ({ videoId }: { videoId: string }) => {
...
+  const descText = selectedVideo ? (selectedVideo as any).snippet.description : 'None';
+  const descPreview = descText.substring(0, 150) + ' ...';
+  const extraContent = (
+    <div>
+      <p className="extra-content">{descText}</p>
+    </div>
+  );
+
+  const moreButtonName = readMore ? 'Collapse' : 'Read More';
+  const isLandingPage = selectedVideo ? selectedVideo.id === '_ok27SPHhwA' : false;
```
Comment (JKHeadley) [inline src/routes/Home/index.tsx:302]: This is okay until we go public, however the `_ok27SPHhwA` video is an actual video so this is really a hack that overrides that video.  We can work on a better method later.

File: src/components/LandingPage.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
```
Comment (JKHeadley) [inline src/components/LandingPage.tsx:1]: Thank you for adding a landing page! As with other of my comments in this PR, I think this is a great addition for volunteer functionality.  However when we go public, this will be confusing as general users will not be using the app as a "timestamping tool" but as more of a "youtube for mbt" where they can search, like videos, create playlists, etc.

**Review by JKHeadley — CHANGES_REQUESTED** [review]
Overall very nice changes. There are a few things that will need to be fixed before we merge this PR.

As a side note, in the future it will probably be better to split PR's into more congruent chunks, rather than mixing small fixes with more "undefined" features.  For example, in this PR most of the changes are incremental improvements to the UI/UX, while other changes are more major and feature-like such as the landing page.

Splitting these into separate PRs will help things move forward more smoothly, since it is more likely that some changes get approved and merged more quickly than others.

**Comment by JKHeadley** [comment]
Just FYI I'm going to go ahead and merge in some semi-major changes that have been coming for a while. Don't worry if there are any conflicts. Once we are ready to merge this branch I will pull in the master branch first to resolve any conflicts.

**Comment by oormicreations** [comment]
@JKHeadley Thank you very much for the reviews and suggestions. I am making the suggested changes. I hope it will make the app better and faster. (after timestamping about 200 videos, I know what slows me down, and so I am trying to fix it).

File: src/routes/Edit/VideoSegmentItem.tsx (hunk 4 above)
Comment (oormicreations) [reply src/routes/Edit/VideoSegmentItem.tsx:36]: This is not possible at this time, there is no "lightgrey" color.

File: src/routes/Home/index.tsx (hunk 9 above)
Comment (oormicreations) [reply src/routes/Home/index.tsx:340]: There is no built-in ui item that matches the "show more" style. So for now, I can not do it without coding a totally new component or CSS class, which is not necessary I guess.

**Comment by oormicreations** [comment]
@JKHeadley I made changes where I could. Thank you. I will send the next batch soon.

**Comment by JKHeadley** [comment]
closes #47

**Review by JKHeadley — APPROVED** [review]
